helpers.py

```diff
@@ -0,0 +1,162 @@
+    h = se * stats.t.ppf((1 + 0.95) / 2., n-1) #compute the 95% confidence interval with the se.
+    all_session_mean  = df.mean()
+    y1  = all_session_mean+h
+    y2 = all_session_mean-h
+    plt.plot(all_session_mean)
+    plt.fill_between(x = np.arange(0,len(all_session_mean)), y1=y1,y2 = y2, alpha = 0.2)
+    plt.xticks(np.arange(0,40,10),labels=np.arange(-2,0,0.5))
+    plt.show()
+####################################################################################################################
+#%%
+def get_behave_data(path_to_parent_folder, choose_outcome = 'all', return_dict_or_df = 'all', include_auto_reward_trials = False):
+
+    from pathlib import Path
+    import json
+    import numpy as np
+    import pandas as pd
+
+    print('path to parent folder:', path_to_parent_folder, ', choose outcome = ', choose_outcome, ', include auto reward trials = ', include_auto_reward_trials, ', return dict or df = ', return_dict_or_df) 
+    path_to_parent_folder
+    choose_outcome
+    include_auto_reward_trials
+
+    
+   
+
+    # 'X:/public/projects/BeJG_20230130_VisDetect/'
+    data_path = Path(path_to_parent_folder) #Define the path to the global parent folder within Ceph shared lab space. 
+    session_settings_paths = list(data_path.rglob("*session_settings.json")) # Get a list of all paths for the session settings. 
+    per_session_trials_paths = list(data_path.rglob("*trials.json")) # Get a list of all paths for the trials conditions for all trials in a session.
+    assert len(per_session_trials_paths) == len(session_settings_paths) # Check that the length of each list is the same (suggesting that each session settings file corresponds to a trials settings file).
+    # Load the data from the json files for trial and session data:
+    all_mouse_session_dict = {'mouse_id':[],'time_stamp':[],'session_trial_outcome':[],'reaction_time':[],'session_TF_vectors':[],
+                              'baseline_TF':[],'change_size_TF':[],'change_time':[],'auto_reward':[]}
+    for session_idx in np.arange(len(session_settings_paths)):
+    # session_idx = 0
+        try:
+            with open(session_settings_paths[session_idx]) as session:
+                session_settings_dict = json.load(session)
+                session_settings = pd.Series(session_settings_dict) # per session settings relevant for all trials within a session.
+            
+            with open(per_session_trials_paths[session_idx]) as per_session_trials: # this is per session (json file includes all the trials that occured in that session)
+                per_session_trials_dict= json.load(per_session_trials)
+                per_session_trials_data = pd.Series(per_session_trials_dict) # output is a series where each element is a dictionary of values for a single trial.
+        except:
+            print('json files are empty for: ', session_settings_paths[session_idx] )
```

Comment:
I wonder if this can be optimised further, e.g. is it possible to not write empty .json files in the first place? or to pre-select known jsons that are not empty based on what mouse / sessions were run?